h/util/metrics.py

```diff
@@ -68,18 +32,24 @@ def record_search_api_usage_metrics(params):
         # Record usage of tags/tag (tags is an alias of tag).
         "tags",
         "tag",
-        # Record usage of _separate_replies which will help distinguish client calls
-        # for loading the sidebar annotations from other api calls.
-        "_separate_replies",
         # Record group and user-these help in identifying slow queries.
         "group",
         "user",
         # Record usage of wildcard feature.
         "wildcard_uri",
     ]
+    # The New Relic Query Language does not permit _ at the begining
+    # and offset is a reserved key word.
+    params = [("es_{}".format(k), params[k]) for k in keys if k in params]
 
-    for k in keys:
-        if k in params:
-            # The New Relic Query Language does not permit _ at the begining
-            # and offset is a reserved key word.
-            send_metric("es_{}".format(k), params[k])
+    # Record usage of _separate_replies which will help distinguish client calls
```

Comment:
It's a private param not exposed to our customers in the api docs. This particular param is a bit odd in that it is passed in via the search query params from the client and then popped off the dict and passed into the Search class init as a config option. I don't like it either but I would like to record whether this is set or not and I didn't want to add a bunch of logic to the Search class to add it back as a key and then remove it after. The alternative is to make a new function specifically for this option-this might be cleaner in some ways but it also requires a whole new function just to record one setting which seems like a bit much. A little history of why this is treated as a setting: It's used to append an extra search filter to only return reply annotations and to not run the search to get replies since the annotations being returned are already replies. All of this is a bit odd and really comes down to the Search class running multiple searches because of the way we implement replies. In my opinion Search shouldn't run multiple searches and this param should really stay a param but obviously that's not the case right now. I don't have a really strong opinion either way, I just want it recorded as it's useful information.